libsolidity/formal/SMTEncoder.cpp

```diff
@@ -2810,11 +2820,12 @@ ModifierDefinition const* SMTEncoder::resolveModifierInvocation(ModifierInvocati
 	return modifier;
 }
 
-vector<FunctionDefinition const*> const& SMTEncoder::contractFunctions(ContractDefinition const& _contract)
+set<FunctionDefinition const*, ASTNode::CompareByID> const& SMTEncoder::contractFunctions(ContractDefinition const& _contract)
```

Comment:
The order in which the functions are visited influences the order in which the verification targets are created, checked and reported. Since before it was a vector, that was fine, but having a set of pointers causes nondeterminism, so it needs to be sorted by something that is consistent throughout runs (the AST ids are)